src/planner/ConfigurationToConfigurations.cpp

```diff
@@ -0,0 +1,82 @@
+#include "aikido/planner/ConfigurationToConfigurations.hpp"
+
+#include "aikido/constraint/Testable.hpp"
+
+namespace aikido {
+namespace planner {
+
+//==============================================================================
+ConfigurationToConfigurations::ConfigurationToConfigurations(
+    statespace::StateSpacePtr stateSpace,
+    const statespace::StateSpace::State* startState,
+    const std::vector<statespace::StateSpace::State*>& goalStates,
+    statespace::InterpolatorPtr interpolator,
+    constraint::TestablePtr constraint)
+  : Problem(std::move(stateSpace))
+  , mStartState(startState)
+  , mGoalStates(goalStates)
+  , mInterpolator(std::move(interpolator))
+  , mConstraint(std::move(constraint))
+{
+  if (mStateSpace != mConstraint->getStateSpace())
+  {
+    throw std::invalid_argument(
+        "StateSpace of constraint not equal to StateSpace of planning space");
+  }
+}
+
+//==============================================================================
+const std::string& ConfigurationToConfigurations::getName() const
+{
+  return getStaticName();
+}
+
+//==============================================================================
+const std::string& ConfigurationToConfigurations::getStaticName()
+{
+  static std::string name("ConfigurationToConfigurations");
+  return name;
+}
+
+//==============================================================================
+const statespace::StateSpace::State*
+ConfigurationToConfigurations::getStartState() const
+{
+  return mStartState;
+}
+
+//==============================================================================
+const std::vector<statespace::StateSpace::State*>
```

Comment:
`const std::vector<const statespace::StateSpace::State*>`

Although like I mentioned, we would want an `addGoal()` so the container cannot be `const` in that case.